servers/physics_3d/godot_collision_solver_3d_sat.cpp

```diff
@@ -877,11 +878,13 @@ static void _collision_sphere_capsule(const GodotShape3D *p_a, const Transform3D
 	const GodotSphereShape3D *sphere_A = static_cast<const GodotSphereShape3D *>(p_a);
 	const GodotCapsuleShape3D *capsule_B = static_cast<const GodotCapsuleShape3D *>(p_b);
 
-	real_t capsule_B_radius = capsule_B->get_radius();
+	real_t scale_A = p_transform_a.basis[0].length();
+	real_t scale_B = p_transform_b.basis[0].length();
+	real_t capsule_B_radius = capsule_B->get_radius() * scale_B;
 
 	// Construct the capsule segment (ball-center to ball-center)
 	Vector3 capsule_segment[2];
-	Vector3 capsule_axis = p_transform_b.basis.get_column(1) * (capsule_B->get_height() * 0.5 - capsule_B_radius);
+	Vector3 capsule_axis = p_transform_b.basis.get_column(1) * (capsule_B->get_height() * scale_B * 0.5 - capsule_B_radius);
```

Comment:
This is double-applying the `p_transform_b` scale. Firstly the scaling is performed by the algebraic `scale_B` and modified `capsule_B_radius`, and then scaled again by multiplying by the scaled `p_transform_b` basis.

I got this to work by:
 1. Restoring the `capsule_B_radius` creation line to original (unscaled)
 2. Restoring the `capsule_axis` creation line to original (scaled once by `p_transform_b.basis` multiply)
 3. Adding the multiply of `capsule_B_radius * scale_B` in the `analytic_sphere_collision` call.